sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -314,6 +314,11 @@ protected string GetOptionalVariable(string name)
                 value = Environment.GetEnvironmentVariable(name);
             }
 
+            if (value == null)
```

Comment:
Which use case does this enable?

Reply:
if you are using the standard env variables for DefaultAzureCredential this allows the test framework to pick those up in RecordMode instead of needing two sets of env variables.  The test env variables are still chosen first so if you have them it will use that as it always did.